packages/dd-trace/src/appsec/iast/overhead-controller.js

```diff
@@ -0,0 +1,104 @@
+'use strict'
+
+const OVERHEAD_CONTROLLER_CONTEXT_KEY = 'oce'
+
+const ANALYZE_REQUEST = 'ANALYZE_REQUEST'
+const REPORT_VULNERABILITY = 'REPORT_VULNERABILITY'
+
+class Quota {
+  constructor (reserved, release) {
+    this._acquired = !!reserved
+    this._operationRelease = release
+    this._released = false
+  }
+
+  isAcquired () {
+    return this._acquired
+  }
+
+  release () {
+    if (this._acquired && !this._released) {
+      this._operationRelease()
+      this._released = true
+    }
+  }
+}
+
+const LONG_RUNNING_OPERATIONS = {
+  ANALYZE_REQUEST: {
+    hasQuota: () => {
+      const reserved = concurrentRequestTokens > 0
```

Comment:
The initial version was targeting just a percentage the requests and then we wanted to move to the concurrent number of requests. In the end, the important thing is that the API is designed in a way that we can plug different strategies, for example:
1. Analyze at most two concurrent requests 
2. Analyze only 30% of the requests
3. Analyze each endpoint only once
4. Allow only two concurrent requests and only 30% of the requests
5. ...